Documentation/ci-platform-coverage-guidelines.md

```diff
@@ -0,0 +1,41 @@
+# Guidelines for Platforms Tested in CI
+
+This document contains the guidelines for which platforms (OS and architectures) to test in
+the source build CI.
+
+## Distro Families
+
+1. Prefer testing base distros of families over derivatives.
+1. Prioritize testing distros that source build .NET.
+1. Use CentOS Stream instead of Red Hat because it is the free alternative.
+1. Include a permutation of distros for the supported C standard library implementations
+(e.g. glibc and musl).
+1. Only test officially supported distros. Community supported distros will not be covered.
```

Comment:
When referring to "supported", it needs to be clarified what that's referring to. I'm assuming it means "supported by .NET". The same thing applies to the Architectures section.

Reply:
Reworded this to be 

```
Only test [distros that are officially supported by .NET](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md#net-supported-os-policy).
Community supported distros will not be covered.
```

I refactored the architecture section to list amd64 and arm64 explicitly.  I decided this after realizing arm32 is officially supported by .NET but not something we support for source-build.